### PR TITLE
refactor greptimedb bridge (split to actions and connectors)

### PR DESCRIPTION
**Pull request by SergeTupchiy**
Fixes EMQX-11457

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: changes/ee/refactor-12386.en.md
Comment (thalesmg) [inline changes/ee/refactor-12386.en.md:1]: I think this won't be included in the formatted changelog with this filename.  Maybe rename it to `feat-...`?

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl [hunk 1]
```diff
@@ -62,6 +70,38 @@
...
+        channels => Channels#{ChannelId => ChannelConfig}
+    }}.
+
+on_remove_channel(_InstanceId, #{channels := Channels} = State, ChannelId) ->
+    NewState = State#{channels => maps:remove(ChannelId, Channels)},
+    {ok, NewState}.
+
+on_get_channel_status(InstanceId, _ChannelId, State) ->
+    case on_get_status(InstanceId, State) of
+        connected -> connected;
```
Comment (thalesmg) [inline apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl:98]: Please use `?status_connected`, `?status_connecting`, etc. from `emqx_resource.hrl`, so we avoid typos and normalize the usage of those macros.

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl [hunk 2]
```diff
@@ -164,6 +214,14 @@ on_get_status(_InstId, #{client := Client}) ->
             disconnected
     end.
 
+transform_bridge_v1_config_to_connector_config(BridgeV1Config) ->
```
Comment (thalesmg) [inline apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl:217]: Nit: maybe move this to the action info module?

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl [hunk 3]
```diff
@@ -29,44 +35,67 @@ conn_bridge_examples(Method) ->
...
-values(common, Protocol, SupportUint, TypeOpts) ->
-    CommonConfigs = #{
-        type => list_to_atom(Protocol),
+connector_examples(Method) ->
+    [
+        #{
+            <<"greptimedb">> => #{
+                summary => <<"GreptimeDB Connector">>,
+                value => emqx_connector_schema:connector_values(
+                    Method, influxdb, connector_values(Method)
```
Comment (thalesmg) [inline apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl:66]:
```suggestion
                    Method, greptimedb, connector_values(Method)
```
?

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl [hunk 4]
```diff
@@ -29,44 +35,67 @@ conn_bridge_examples(Method) ->
...
-                "bool=${payload.bool}">>,
+        write_syntax => write_syntax_value(),
         precision => ms,
         resource_opts => #{
             batch_size => 100,
             batch_time => <<"20ms">>
         },
+        username => <<"example_username">>,
+        password => <<"******">>,
+        dbname => <<"exampl_db">>,
```
Comment (thalesmg) [inline apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl:86]:
```suggestion
        dbname => <<"example_db">>,
```

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl (hunk 1 above)
Comment (SergeTupchiy) [reply apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl]: Done, thanks!

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl (hunk 2 above)
Comment (SergeTupchiy) [reply apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb_connector.erl]: I implemented this function following influxdb connector but then realized that it was needed only for some influxdb tests but forgot to remove it in greptimedb bridge... Now I've removed it, thanks. Note: greptimedb action module already has a similar function.

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl (hunk 3 above)
Comment (SergeTupchiy) [reply apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl]: :see_no_evil: fixed, thanks!

File: apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl (hunk 4 above)
Comment (SergeTupchiy) [reply apps/emqx_bridge_greptimedb/src/emqx_bridge_greptimedb.erl]: Done :see_no_evil:

File: changes/ee/refactor-12386.en.md
Comment (SergeTupchiy) [reply changes/ee/refactor-12386.en.md:1]: Thanks, fixed.. :see_no_evil:

**Review by thalesmg — COMMENTED** [review]
:heavy_check_mark: 
LGTM.
We just need to update the dependency ref once the PR is merged (or use our fork if that takes too long).

**Comment by SergeTupchiy** [comment]
Thanks @thalesmg.  Greptime team approved my PR and tagged the new release, so I've already removed my fork from rebar/mix deps.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**